lib/datadog/ci/ext/environment.rb

```diff
@@ -156,7 +163,10 @@ def extract_buildkite(env)
             TAG_PIPELINE_URL => env['BUILDKITE_BUILD_URL'],
             TAG_JOB_URL => "#{env['BUILDKITE_BUILD_URL']}##{env['BUILDKITE_JOB_ID']}",
             TAG_PROVIDER_NAME => 'buildkite',
-            TAG_WORKSPACE_PATH => env['BUILDKITE_BUILD_CHECKOUT_PATH']
+            TAG_WORKSPACE_PATH => env['BUILDKITE_BUILD_CHECKOUT_PATH'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_NAME => env['BUILDKITE_BUILD_AUTHOR'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_EMAIL => env['BUILDKITE_BUILD_AUTHOR_EMAIL'],
+            Datadog::Ext::Git::TAG_COMMIT_MESSAGE => env['BUILDKITE_MESSAGE']
           }
```

Comment:
Minor: Is it just me, or are these huge hashes a domain object wanting to jump out?

```ruby
EnvironmentInfo.new(
  tag_branch: ...,
  tag_commit_sha: ...,
  #...
)
```

There's even additional normalization in the `tags` top-level function that would look to be something that we did in the `#initialize` of this domain object.

Reply:
I agree.
BUT, I'm asking for rain cheque and I'll address this in a refactor after I finish this CI spec update and the local .git environment sourcing.